src/network/networktool.cpp

```diff
@@ -38,19 +38,38 @@ void NetworkTool::checkNewVersionInGitHubPayload(QJsonDocument& jsonDocument)
 {
     const static QString projectTag = QStringLiteral(GIT_TAG);
 
+    auto semverToInt = [](const QString& version) -> int {
+        int result = 0;
+        QStringList subVersion = version.right(1).split('.');
+
+        // Check if subVersion is valid
+        if(subVersion.length() > 3)
+        {
+            return result;
+        }
+
+        // Transform version to int
+        for(int i = 0; i < subVersion.length(); i++)
+        {
+            result += subVersion[i].toInt()<<(3-i);
```

Comment:
This is not right at all. How did you come up with this?

version 1.2.3 is equivalent to 2.1.1

version 1.11.0 is greater than version 2.0.0

Reply:
Can we do something analogous to this?
https://github.com/mavlink/qgroundcontrol/pull/6913/files#diff-ae5366dfccfee1039895edb6b2131a6eR731